template/public/app.js

```diff
@@ -1,34 +1,48 @@
-import moment from 'moment';
-import { uiModules } from 'ui/modules';
-import uiRoutes from 'ui/routes';
+import React from "react";
+import moment from "moment";
+import { uiModules } from "ui/modules";
+import chrome from "ui/chrome";
+import uiRoutes from "ui/routes";
+import { render, unmountComponentAtNode } from "react-dom";
 
-import 'ui/autoload/styles';
-import './less/main.less';
-import template from './templates/index.html';
+import "ui/autoload/styles";
+import "./less/main.less";
+import template from "./templates/index.html";
+import { Main } from "./components/main";
 
+const app = uiModules.get("apps/<%= name %>");
+app.config(stateManagementConfigProvider => {
+  stateManagementConfigProvider.disable();
+});
+const controller = ($scope, $route, $interval) => {
+  const elem = document.getElementById("pluginReactRoot");
```

Comment:
I see what you're doing here with the template and such. This isn't how any of the other React apps work, as far as I know. There's two methods I've seen in use so far, a react directive, and using Kibana's root controller. Here's how the latter works:

```js
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import chrome from 'ui/chrome';
import { Main } from './components/main/main';

function RootController($scope, $element) {
  const domNode = $element[0];

  // render react to DOM
  render(<Main />, domNode);

  // unmount react on controller destroy
  $scope.$on('$destroy', () => {
    unmountComponentAtNode(domNode);
  });
}

chrome.setRootController('myplugin', RootController);
```

You can drop this into the top-level `app.js` file, and then kick off the actual application in `components/App.js`. Paths and naming can all be changed here, but this separates the angular bootstrapping from the react app itself. This also means we can render the react app without using the angular router, and can avoid enabling that entirely.

@spalger any additional input here?

Reply:
Also, if we're in favor of not using the angular router, we should also disable the location provider:

```js
app.config($locationProvider => {
  $locationProvider.html5Mode({
    enabled: false,
    requireBase: false,
    rewriteLinks: false,
  });
});
```

Otherwise it'll rewrite links by default and break whatever other router is being used.